packages/dbml-core/src/parse/dbml/src/lib/lexer/lexer.ts

```diff
@@ -0,0 +1,567 @@
+          break;
+        case ':':
+          this.advance();
+          this.addToken(SyntaxTokenKind.COLON);
+          break;
+        case '<':
+          this.advance();
+          if (this.peek() !== '=') {
+            this.addToken(SyntaxTokenKind.LT);
+          } else {
+            this.advance();
+            this.addToken(SyntaxTokenKind.LE);
+          }
+          break;
+        case '>':
+          this.advance();
+          if (this.peek() !== '=') {
+            this.addToken(SyntaxTokenKind.GT);
+          } else {
+            this.advance();
+            this.addToken(SyntaxTokenKind.GE);
+          }
+          break;
+        case '=':
+          this.advance();
+          if (this.peek() !== '=') {
+            this.addToken(SyntaxTokenKind.EQUAL);
+          } else {
+            this.advance();
+            this.addToken(SyntaxTokenKind.DOUBLE_EQUAL);
+          }
+          break;
+        case '!':
+          this.advance();
+          if (this.peek() !== '=') {
+            this.addToken(SyntaxTokenKind.EXCLAMATION);
+          } else {
+            this.advance();
+            this.addToken(SyntaxTokenKind.NOT_EQUAL);
+          }
+          break;
+        case "'":
+          if (this.checkTripleQuote()) {
+            this.multilineStringLiteral();
+          } else {
+            this.singleLineStringLiteral();
+          }
+          break;
+        case '"':
+          this.quotedVariable();
+          break;
+        case '`':
+          this.functionExpression();
+          break;
+        case '#':
+          this.colorLiteral();
+          break;
+        default:
+          if (isAlpha(c)) {
+            this.identifier();
+            break;
+          } else if (isDigit(c)) {
+            this.numericLiteral();
+            break;
+          } else {
+            this.advance();
+            this.errors.push(
+              new ParsingError(
+                ParsingErrorCode.EXPECTED_THINGS,
+                `Unexpected token ${c}`,
+                this.start,
+                this.current - 1,
+              ),
+            );
+            break;
+          }
+      }
+      this.closeGap();
+    }
+    this.tokens.push(SyntaxToken.create(SyntaxTokenKind.EOF, this.start, 0));
+    this.gatherTrivia();
+
+    return new Result(this.tokens, this.errors);
+  }
+
+  gatherTrivia() {
+    let startOfLine = true;
+    let triviaList: SyntaxToken[] = [];
+    let lastNonTrivia: SyntaxToken | undefined;
+    const newTokenList: SyntaxToken[] = [];
+    // eslint-disable-next-line no-restricted-syntax
+    for (const token of this.tokens) {
+      if (isTriviaToken(token)) {
+        triviaList.push(token);
+        if (token.kind === SyntaxTokenKind.NEWLINE) {
+          if (lastNonTrivia) {
+            lastNonTrivia.trailingTrivia = triviaList;
+            startOfLine = true;
+            lastNonTrivia = undefined;
+            triviaList = [];
+          }
+        }
+        continue;
```

Comment:
this code block has 3 nested if with a continue statement. I think the logic is quite hard to follow and understand what's going on here. It's great if you could add some notes for this, or refactor to a simpler logic if possible